### PR TITLE
Add missing CLI environment variables

### PR DESCRIPTION
**Pull request by mikhailshilkov**
### Proposed changes

Apparently, we miss many environment variables in our docs. I added all the ones that I could find (with Claude's help).

### CODE REVIEW

**Review by cnunciato — APPROVED** [review]
Thanks @mikhailshilkov!

Do you know if it's possible to generate these programmatically? It'd be great to not have to remember to do this.

File: content/docs/cli/environment-variables.md [hunk 1]
```diff
@@ -12,6 +12,17 @@ aliases:
...
+    <dt>
+        <span class="font-mono">
+            PULUMI_AI_SERVICE_ENDPOINT
+        </span>
+    </dt>
+    <dd>
+        <p>
+            Specifies the endpoint for Pulumi AI service.
+        </p>
+        <pre><code class="text-xs">PULUMI_AI_SERVICE_ENDPOINT="https://ai.pulumi.com"</code></pre>
```
Comment (cnunciato) [inline content/docs/cli/environment-variables.md:24]:
```suggestion
        <pre><code class="text-xs">PULUMI_AI_SERVICE_ENDPOINT="https://api.pulumi.com"</code></pre>
```

Comment (cnunciato) [inline content/docs/cli/environment-variables.md:24]: Actually .. is this URL a thing? Doesn't seem to resolve for me.

File: content/docs/cli/environment-variables.md [hunk 2]
```diff
@@ -111,6 +155,127 @@ aliases:
...
         <pre><code class="text-xs">PULUMI_DEBUG_PROMISE_LEAKS=true</code></pre>
     </dd>
+    <dt>
+        <span class="font-mono">
+            PULUMI_DEV
+        </span>
+    </dt>
+    <dd>
+        <p>
+            Enable features for hacking on pulumi itself.
```
Comment (cnunciato) [inline content/docs/cli/environment-variables.md:165]:
```suggestion
            Enable features for hacking on Pulumi itself.
```

File: content/docs/cli/environment-variables.md [hunk 3]
```diff
@@ -111,6 +155,127 @@ aliases:
...
+        <pre><code class="text-xs">PULUMI_DISABLE_AUTOMATIC_PLUGIN_ACQUISITION=true</code></pre>
+    </dd>
+    <dt>
+        <span class="font-mono">
+            PULUMI_DISABLE_PROVIDER_PREVIEW
+        </span>
+    </dt>
+    <dd>
+        <p>
+            Disables provider preview and enabled previous more concervative preview behavior.
```
Comment (cnunciato) [inline content/docs/cli/environment-variables.md:187]:
```suggestion
            Disables provider preview and enables previous more conservative preview behavior.
```

Incidentally, what does "more conservative" mean"? Might be good to explain this a little.

File: content/docs/cli/environment-variables.md [hunk 4]
```diff
@@ -175,6 +374,28 @@ aliases:
...
+        <pre><code class="text-xs">PULUMI_IGNORE_AMBIENT_PLUGINS=true</code></pre>
+    </dd>
+    <dt>
+        <span class="font-mono">
+            PULUMI_PLUGIN_DOWNLOAD_URL_OVERRIDES
+        </span>
+    </dt>
+    <dd>
+        <p>
+            Specifies overrides for Plugin Download URLs. The expected format is `regexp=URL`, and multiple pairs can be specified separated by commas.
```
Comment (cnunciato) [inline content/docs/cli/environment-variables.md:395]:
```suggestion
            Specifies overrides for plugin-download URLs. The expected format is <code>regexp=URL</code>, and multiple pairs can be specified separated by commas.
```

File: content/docs/cli/environment-variables.md [hunk 5]
```diff
@@ -111,6 +155,127 @@ aliases:
...
+    </dt>
+    <dd>
+        <p>
+            Disables provider preview and enabled previous more concervative preview behavior.
+        </p>
+        <pre><code class="text-xs">PULUMI_DISABLE_PROVIDER_PREVIEW=true</code></pre>
+    </dd>
+    <dt>
+        <span class="font-mono">
+            PULUMI_DISABLE_RESOURCE_REFERENCES
```
Comment (justinvp) [inline content/docs/cli/environment-variables.md:193]: I don't think we need this anymore, so not sure we should even document it. Rather, we should clean up its usage.

File: content/docs/cli/environment-variables.md [hunk 6]
```diff
@@ -111,6 +155,127 @@ aliases:
...
+    </dt>
+    <dd>
+        <p>
+            Disables resource references in Invoke gRPC calls.
+        </p>
+        <pre><code class="text-xs">PULUMI_DISABLE_RESOURCE_REFERENCES=true</code></pre>
+    </dd>
+    <dt>
+        <span class="font-mono">
+            PULUMI_DISABLE_OUTPUT_VALUES
```
Comment (justinvp) [inline content/docs/cli/environment-variables.md:204]: Similar to `PULUMI_DISABLE_RESOURCE_REFERENCES`, though I'm not sure we're quite ready to get rid of it. I don't think we're going to want to maintain this long term, so not sure we should doc it here.

**Pull request merged**